task/maven/0.3/maven.yaml

```diff
@@ -0,0 +1,153 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: maven
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Build Tools
+    tekton.dev/tags: build-tool
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This Task can be used to run a Maven build. It uses a workspace to store m2 local repo.
+
+  workspaces:
+    - name: source
+      description: The workspace consisting of maven project.
+    - name: maven-settings
+      description: >-
+        The workspace consisting of the custom maven settings
+        provided by the user.
+    - name: maven-local-repo
+      description: Local repo (m2) workspace
```

Comment:
I believe we should mark this workspace as optional. Also I think `affinity-assistant` won't allow us to bind two PVC at the same time for a particular task so we might have to document this in readme that in order to bind two pvc we need to disable `affinity-assistant`

Reply:
Hi @vinamra28 .. thanks for the fast reply ... I've tested the solution using an OpenShift cluster, didn't use the `affinity-assistant`, and all worked fine. I would like to understand more how we can test the whole `affinity-assistant` thing. Can I continue testing on a OpenShift cluster? Is the `affinity-assistant` just a good practice ?

Reply:
About the optional workspace, I'm working on that.

Reply:
Where should I document about the affinity-assistant ?

Reply:
I've implemented the optional workspace. How can I aggregate this new commit to the PR?

Reply:
I'm new to all this ... can someone help me ?

Reply:
hey @jpmaida, apologies for replying late here

>  I've tested the solution using an OpenShift cluster, didn't use the affinity-assistant, and all worked fine. I would like to understand more how we can test the whole affinity-assistant thing.

by default openshift disabled the `affinity-assistant` hence you were able to use that easily. `tektoncd/pipelines` by default enables the `affinity-assitant` which users have to disable manually

> Can I continue testing on a OpenShift cluster?

Are you using openshift-pipelines ? If so, you need to disable that using `TektonConfig`. @vdemeester @piyush-garg please correct me if I am wrong

Reply:
> Where should I document about the affinity-assistant ?

In the readme of the task

Reply:
Done. How do we proceed?

Reply:
?

Reply:
@vinamra28 ?